Documentation for logprob module (#6858)
pymc-devs/pymc: docs/source/api/logprob.rst

   conditional_logp
   transformed_conditional_logp

Derived probability functions
-----------------------------

> In PyMC users can derive their own custom distributions. Custom distribution refers to the ability to define and use probability distributions that are not included in the standard set of distributions provided.

Reviewer: ```suggestion
In PyMC, users can create their own custom distributions, which refers to the ability to define and utilize probability distributions not found in the standard set provided.
```

> While PyMC provides a wide range of common probability distributions (e.g., Normal, Bernoulli, etc.), there may be cases where you need to use a distribution that is not available by default. In such cases, you can create your own custom distribution using the :class:`~pm.CustomDist` class provided by PyMC.

Reviewer 2: ```suggestion
While PyMC provides a wide range of common probability distributions (e.g., Normal, Bernoulli, etc.), there may be cases where you need to use a distribution that is not available by default. In such cases, you can create your own custom distribution using the :class:`~pymc.CustomDist` class provided by PyMC.
```

The "name" used for the hyperlink should be the whole **import path** where it is documented (not where it is found in the codebase). `pm` is therefore not available. To shorten it, the `~` is available, so the text only shows the object and skips the library and modules (if any).

> While PyMC provides a wide range of common probability distributions (e.g., Normal, Bernoulli, etc.), there may be cases where you need to use a distribution that is not available by default. In such cases, you can create your own custom distribution using the pm.CustomDist class provided by PyMC.

Reviewer: ```suggestion
While PyMC provides a wide range of common probability distributions (e.g., Normal, Bernoulli, etc.), there may be cases where you need to use a distribution that is not available by default. In such cases, you can create your own custom distribution using the :class:`~pm.CustomDist` class provided by PyMC.
```

> Simplest way to define a Custom Distribution can be better understood from the following example:

Reviewer: ```suggestion
The use of the :class:`~pm.CustomDist` API can be better understood from the following example:
```
There are two ideas combined into one that is not intuitively phrased. Either go "The simplest way to define a custom distribution in PyMC is..." or "Using pm.CustomDist can be illustrated with...". I prefer the latter

.. code-block:: python

    import pymc as pm
    from pytensor.tensor import TensorVariable
    def dist(
        lam: TensorVariable,
        shift: TensorVariable,
        size: TensorVariable,
    ) -> TensorVariable:
        return pm.Exponential.dist(lam, size=size) + shift
    with pm.Model() as m:
        lam = pm.HalfNormal("lam")
        shift = -1
        pm.CustomDist(
    	"custom_dist",
    	lam,
    	shift,
    	dist=dist,
    	observed=[-1, -1, 0],
        )
        prior = pm.sample_prior_predictive()
        posterior = pm.sample()

> Here, we provide a dist function that creates a PyTensor graph built from other PyMC distributions. PyMC can automatically infer that the logp of this variable corresponds to a shifted Exponential distribution.

Reviewer: ```suggestion
Here, we build a PyTensor graph from other PyMC distributions through :class:`~pymc.CustomDist`. PyMC can automatically infer that the logp of this graph that corresponds to a shifted Exponential distribution.
```

> For more details, check out:
> 
> .. currentmodule:: pymc
> .. autosummary::
>    :toctree: generated/
> 
>    CustomDist

Reviewer 2: CustomDist already has API docs. This would repeat them here.

```suggestion
```

It has already been linked above, so I think we are good. If we still want to emphasize the customdist relation, we can use a `seealso` directive.
